File: 4. Maximum And Minimum In Linked List - GFG/4.-maximum-and-minimum-in-linked-list.py
#User function Template for python3

'''
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
'''
def maximum(head):
    #code here
    current = head
    maxi = current.data
    while(current != None):
        if(current.data > maxi ):
            maxi = current.data
        current = current.next
    return maxi
    
    
def minimum(head):
    #code here
    current = head
    mini = current.data
    while(current != None):
        if(current.data < mini):
            mini = current.data
        current = current.next
    return mini

#{ 
#  Driver Code Starts
#Initial Template for Python 3

import io
import sys

    
# Node class
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
#Linked list class
class LinkedList:
    def __init__(self):
        self.head=None
        self.tail=None

    # append at the end of the list
    def append(self,new_node):
        if self.head is None:
            self.head=new_node
            self.tail=new_node
            return
        self.tail.next=new_node
        self.tail = self.tail.next

if __name__ == '__main__':
    t=int(input())
    for cases in range(t):
        n=int(input())
        a=LinkedList()
        nodes=list(map(int,input().strip().split())) #list containing nodes
        for x in nodes:
            node=Node(x) # create a new node
            a.append(node)
        print(maximum(a.head), end = " ")
        print(minimum(a.head))
# } Driver Code Ends

File: 4. Maximum And Minimum In Linked List - GFG/4.-maximum-and-minimum-in-linked-list.cpp
// { Driver Code Starts
#include <bits/stdc++.h> 
using namespace std; 

struct Node
{
    int data;
    struct Node* next;
    
    Node(int x){
        data = x;
        next = NULL;
    }
};

int maximum(struct Node* head);
int minimum(struct Node* head);

int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;

        int data;
        cin>>data;
        struct Node *head = new Node(data);
        struct Node *tail = head;
        for (int i = 0; i < n-1; ++i)
        {
            cin>>data;
            tail->next = new Node(data);
            tail = tail->next;
        }
        cout << maximum(head) << " " << minimum(head) << endl;
    }
    return 0;
}// } Driver Code Ends


//User function Template for C++

/*
struct Node
{
    int data;
    Node * next;
    Node (int x)
    {
        data=x;
        next=NULL;
    }
        
};
*/

int maximum(Node *head)
{
    //Your code here
    int max = INT_MIN;
    Node *current = head;
    while(current != NULL){
        if (current->data > max){
            max = current->data;
        }
        current = current->next;
    }
    return max;
}

int minimum(Node *head)
{
    //Your code here
    int min = INT_MAX;
    Node *current = head;
    while(current != NULL){
        if (current->data < min){
            min = current->data;
        }
        current = current->next;
    }
    return min;
}